From a7c46c1204f210aebe67afb703b8490a67517e65 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Tue, 8 Oct 2019 12:56:02 +0300
Subject: [PATCH] arm64: dts: ls2088a: use a pseudo-bus to constrain usb dma
 size

commit 3c094fe29944a0a3aefe5c8fb80ada5f399e4b92 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Wrap the usb controllers in an intermediate simple-bus and use it to
constrain the dma address size of these usb controllers to the 40 bits
that they generate toward the interconnect.

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls208xa.dtsi | 51 +++++++++++--------
 1 file changed, 29 insertions(+), 22 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
index 42d04a2e2b34..a6042f4b7a27 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls208xa.dtsi
@@ -723,28 +723,35 @@
 			dma-coherent;
 		};
 
-		usb0: usb3@3100000 {
-			status = "disabled";
-			compatible = "fsl,ls2088a-dwc3", "snps,dwc3";
-			reg = <0x0 0x3100000 0x0 0x10000>;
-			interrupts = <0 80 0x4>; /* Level high type */
-			dr_mode = "host";
-			snps,quirk-frame-length-adjustment = <0x20>;
-			snps,dis_rxdet_inp3_quirk;
-			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
-			snps,host-vbus-glitches;
-		};
-
-		usb1: usb3@3110000 {
-			status = "disabled";
-			compatible = "fsl,ls2088a-dwc3", "snps,dwc3";
-			reg = <0x0 0x3110000 0x0 0x10000>;
-			interrupts = <0 81 0x4>; /* Level high type */
-			dr_mode = "host";
-			snps,quirk-frame-length-adjustment = <0x20>;
-			snps,dis_rxdet_inp3_quirk;
-			snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
-			snps,host-vbus-glitches;
+               aux_bus: aux_bus {
+                       #address-cells = <2>;
+                       #size-cells = <2>;
+                       compatible = "simple-bus";
+                       ranges;
+                       dma-ranges = <0x0 0x0 0x0 0x0 0x100 0x00000000>;
+
+                       usb0: usb3@3100000 {
+                               status = "disabled";
+                               compatible = "snps,dwc3";
+                               reg = <0x0 0x3100000 0x0 0x10000>;
+                               interrupts = <0 80 0x4>; /* Level high type */
+                               dr_mode = "host";
+                               snps,quirk-frame-length-adjustment = <0x20>;
+                               snps,dis_rxdet_inp3_quirk;
+                               snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+                               snps,host-vbus-glitches;
+                       };
+                       usb1: usb3@3110000 {
+                               status = "disabled";
+                               compatible = "snps,dwc3";
+                               reg = <0x0 0x3110000 0x0 0x10000>;
+                               interrupts = <0 81 0x4>; /* Level high type */
+                               dr_mode = "host";
+                               snps,quirk-frame-length-adjustment = <0x20>;
+                               snps,dis_rxdet_inp3_quirk;
+                               snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
+                               snps,host-vbus-glitches;
+                       };
 		};
 
 		ccn@4000000 {
-- 
2.17.1

